lib/text/ttml_text_parser.js

```diff
@@ -625,29 +726,41 @@ shaka.text.TtmlTextParser = class {
    * @param {string} attributeName
    * @param {!Array.<Element>} collection
    * @param {string} prefixName
-   * @return {Element}
+   * @return {Array.<!Element>}
    * @private
    */
   static getElementFromCollection_(
       element, attributeName, collection, prefixName) {
+    const items = [];
+
     if (!element || collection.length < 1) {
-      return null;
+      return items;
     }
-    let item = null;
-    const itemName = shaka.text.TtmlTextParser.getInheritedAttribute_(
+
+    const attributeValue = shaka.text.TtmlTextParser.getInheritedAttribute_(
         element, attributeName);
-    if (itemName) {
-      for (let i = 0; i < collection.length; i++) {
-        if ((prefixName + collection[i].getAttribute('xml:id')) == itemName) {
-          item = collection[i];
-          break;
+
+    if (attributeValue) {
+      // There could be multiple items in one attribute
+      // <span style="style1 style2">A cue</span>
+      const itemNames = attributeValue.split(' ');
+
+      for (let i = 0; i < itemNames.length; i++) {
```

Comment:
style nit: use for of loops instead:
for (name of itemNames)
 for (item in collection)